web/src/routes/auth/pin-prompt/+page.ts

```diff
@@ -0,0 +1,27 @@
+import { AppRoute } from '$lib/constants';
+import { authenticate } from '$lib/utils/auth';
+import { getFormatter } from '$lib/utils/i18n';
+import { getAuthStatus } from '@immich/sdk';
+import { redirect } from '@sveltejs/kit';
+import type { PageLoad } from './$types';
+
+export const load = (async ({ url }) => {
+  await authenticate();
+
+  const { isElevated, pinCode } = await getAuthStatus();
+
+  const continuePath = url.searchParams.get('continue');
+  if (isElevated) {
+    redirect(302, AppRoute.LOCKED);
+  }
```

Comment:
this is wrong, if you are already elevated you should be forwarded to the continue path.